host/bcm283x/gpio.go

```diff
@@ -80,8 +82,24 @@ var (
 // Present returns true if running on a Broadcom bcm283x based CPU.
 func Present() bool {
 	if isArm {
+		// This checks a magic string which the
+		// https://github.com/raspberrypi/linux kernel patched into
+		// /proc/cpuinfo.
 		hardware, ok := distro.CPUInfo()["Hardware"]
-		return ok && strings.HasPrefix(hardware, "BCM")
+		if ok && strings.HasPrefix(hardware, "BCM") {
+			return true
+		}
+		// For non-Raspbian systems (e.g. gokrazy, Fedora, Debian, …):
+		b, err := ioutil.ReadFile("/proc/device-tree/compatible")
```

Comment:
Please use distro.DTCompatible() so it's only read once per process.

Reply:
Done, thanks for the pointer!